src/Domain/Creators/TimeDependence/CubicScale.cpp

```diff
@@ -0,0 +1,130 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Creators/TimeDependence/CubicScale.hpp"
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/CubicScale.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+
+namespace domain {
+namespace creators {
+namespace time_dependence {
+template <size_t MeshDim>
+CubicScale<MeshDim>::CubicScale(
+    const double initial_time, const double outer_boundary,
+    std::array<std::string, 2> functions_of_time_names,
+    const std::array<double, 2>& initial_expansion,
+    const std::array<double, 2>& velocity,
+    const std::array<double, 2>& acceleration) noexcept
+    : initial_time_(initial_time),
+      outer_boundary_(outer_boundary),
+      functions_of_time_names_(std::move(functions_of_time_names)),
+      initial_expansion_(initial_expansion),
+      velocity_(velocity),
+      acceleration_(acceleration) {}
+
+template <size_t MeshDim>
+std::unique_ptr<TimeDependence<MeshDim>>
+CubicScale<MeshDim>::get_clone() const noexcept {
+  return std::make_unique<CubicScale>(
+      initial_time_, outer_boundary_, functions_of_time_names_,
+      initial_expansion_, velocity_, acceleration_);
+}
+
+template <size_t MeshDim>
+std::vector<std::unique_ptr<
+    domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, MeshDim>>>
+CubicScale<MeshDim>::block_maps(const size_t number_of_blocks) const noexcept {
+  ASSERT(number_of_blocks > 0, "Must have at least one block to create.");
+  std::vector<std::unique_ptr<
+      domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, MeshDim>>>
+      result{number_of_blocks};
+  result[0] = std::make_unique<MapForComposition>(map_for_composition());
+  for (size_t i = 1; i < number_of_blocks; ++i) {
+    result[i] = result[0]->get_clone();
+  }
+  return result;
+}
+
+template <size_t MeshDim>
+std::unordered_map<std::string,
+                   std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+CubicScale<MeshDim>::functions_of_time() const noexcept {
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      result{};
+  result[functions_of_time_names_[0]] =
+      std::make_unique<FunctionsOfTime::PiecewisePolynomial<3>>(
```

Comment:
Yes, somewhat stupidly because for control systems we would want to them to have a 3rd derivative. I'm not sure if that's a good or bad reason to have it :man_shrugging: @kidder thoughts?

Reply:
I don't know whether or not it makes sense to choose via input file; maybe for now just a comment on why 3rd deriv is chosen